orchestrator/db/helpers.py

```diff
@@ -1,7 +1,27 @@
+import functools
+
+import structlog
+from sqlalchemy import text
 from sqlalchemy.dialects import postgresql
 from sqlalchemy.sql.elements import CompilerElement
 
+from orchestrator.db import db
+
+logger = structlog.get_logger(__name__)
+
 
 def to_sql_string(stmt: CompilerElement) -> str:
     dialect = postgresql.dialect()  # type: ignore
     return str(stmt.compile(dialect=dialect, compile_kwargs={"literal_binds": True}))
+
+
+@functools.cache
+def get_postgres_version() -> int:
+    """Returns the Postgres major version as an int."""
+    try:
+        # The pg_version_num is pg_major_version * 10000 + pg_minor_version
+        pg_version_num = int(db.session.scalar(text("show server_version_num")))
+        return pg_version_num // 10000
+    except ValueError:
+        logger.error("Unable to get query Postgres version")
```

Comment:
Just 'get' or 'query'

Reply:
I don't think that works. `SHOW server_version_num` is the entire sql query. Not a select from a table

Reply:
I think he referred to the wording of the log statement

Reply:
Aha :sweat_smile: